### ISSUE TITLE
Gradle build fails to generate a Persistable

### ISSUE THREAD

**Issue by vrish88** [xmolecules/jmolecules-integrations#42]
Hello there! I ran into a problem trying to get jmolecules bytebuddy working with gradle. It seems this line is to blame https://github.com/xmolecules/jmolecules-integrations/blob/main/jmolecules-bytebuddy/src/main/java/org/jmolecules/bytebuddy/PersistableImplementor.java#L142. The `target` directory is a maven specific output dir and gradle projects won't have them.

Steps to reproduce:
1. Generated a [spring app using gradle](https://start.spring.io/#!type=gradle-project&language=java&platformVersion=2.4.5.RELEASE&packaging=jar&jvmVersion=11&groupId=com.example&artifactId=demo&name=demo&description=Demo%20project%20for%20Spring%20Boot&packageName=com.example.demo&dependencies=lombok,data-jpa,h2)
2. Added the following as the configuration for jmolecules at the top of the `build.gradle`:
    ```groovy
    buildscript {
	    dependencies {
		    classpath platform('org.jmolecules:jmolecules-bom:2021.0.1')
		    classpath 'org.jmolecules.integrations:jmolecules-spring'
		    classpath 'org.jmolecules.integrations:jmolecules-jpa'
		    classpath 'org.jmolecules.integrations:jmolecules-bytebuddy'
	    }
    }
    plugins {
	    id 'java'
	    id 'org.springframework.boot' version '2.4.5'
	    id 'io.spring.dependency-management' version '1.0.11.RELEASE'
	    id 'net.bytebuddy.byte-buddy-gradle-plugin' version '1.11.0'
    }
    
    byteBuddy {
	    transformation {
		    plugin = org.jmolecules.bytebuddy.JMoleculesPlugin
	    }
    }
    ```
3. Add an aggregate root
    ```java
    package com.example.demo;
    
    import lombok.Getter;
    import lombok.Value;
    import org.jmolecules.ddd.types.AggregateRoot;
    import org.jmolecules.ddd.types.Identifier;
    
    import java.util.UUID;
    
    @Getter
    public class Post implements AggregateRoot<Post, Post.PostId> {
        private final PostId id;
    
        public Post() {
            id = PostId.of(UUID.randomUUID());
        }
    
        @Value(staticConstructor = "of")
        public static class PostId implements Identifier {
            UUID id;
        }
    }

    ```
3. Ran `./gradlew assemble`

Got the following error:
```
org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':byteBuddy'.
...
Caused by: java.lang.IllegalStateException: {class com.example.demo.Post=[java.lang.IllegalArgumentException: Could not create directory: target/classes/com/example/demo]} type transformations have failed
	at net.bytebuddy.build.gradle.AbstractByteBuddyTask.doApply(AbstractByteBuddyTask.java:404)
	at net.bytebuddy.build.gradle.ByteBuddyTask.apply(ByteBuddyTask.java:133)`
```

**Comment by vrish88** [comment]
Ah, after digging into it a little further it appears that some of my dependencies were off. I also had this elsewhere in my `build.gradle`

```groovy
dependencies {
	implementation platform('org.jmolecules:jmolecules-bom:2021.0.1')
	implementation 'org.jmolecules.integrations:jmolecules-spring'
	implementation 'org.jmolecules.integrations:jmolecules-jpa'
	implementation 'org.jmolecules.integrations:jmolecules-bytebuddy'
}
```

Once I remove `implementation 'org.jmolecules.integrations:jmolecules-bytebuddy'` it works!

So perhaps, would be project be amenable to some documentation for gradle projects?

**Comment by odrotbohm** [comment]
Thanks for the catch, @vrish88! Admittedly I haven't checked the code generation integration with Gradle at all. Do you have a sample project I could look at? Looks like we need to forward this to ByteBuddy in general as we're not dealing with file system operations ourselves but only through the type creation API of BB.

**Comment by vrish88** [comment]
@odrotbohm If you'd like to play with it I pushed up a sample repo here: https://github.com/vrish88/jmolecules-gradle

**Comment by theborakompanioni** [comment]
I had a similar issue as @vrish88 on my first attempt but this thread helped to get it working
(same error as above regarding "target" dir and really thought it is a library problem at first).

Then I had been facing the problem that the only way to make it work in a more advanced multimodule gradle project
was to disable parallel builds (`org.gradle.parallel=false`) - otherwise bytebuddy runs into a file lock problem.
See https://github.com/raphw/byte-buddy/issues/1054 (relevant issue in bytebuddy-gradle plugin).
This more than triples the time it takes to build the project.

Downgrading net.bytebuddy.byte-buddy-gradle-plugin from v1.11.2 to v1.10.22 resolved to issue and
it seems that after some trouble - it is finally working as expected in a more complex gradle build too!

A small guideline section or even small example project for gradle projects seems like a really helpful thing.
I guess multimodule gradle projects running in parallel mode is very common nowadays.
@odrotbohm What do you think?

**Comment by odrotbohm** [comment]
This should be fixed for the 0.4 snapshots.
